homeassistant/components/mikrotik/__init__.py

```diff
@@ -1 +1,220 @@
+            load_platform(hass, DEVICE_TRACKER, DOMAIN, None, config)
+
+    if not hass.data[DOMAIN][HOSTS]:
+        return False
+    return True
+
+
+class MikrotikClient:
+    """Handle all communication with the Mikrotik API."""
+
+    def __init__(self, host, use_ssl, port, user, password, login_method, encoding):
+        """Initialize the Mikrotik Client."""
+        self._host = host
+        self._use_ssl = use_ssl
+        self._port = port
+        self._user = user
+        self._password = password
+        self._login_method = login_method
+        self._encoding = encoding
+        self._host_name = ""
+        self._info = None
+        self._client = None
+        self._connecting = False
+        self._connected = False
+
+    def connect_to_device(self):
+        """Connect to Mikrotik device."""
+        if self._connecting:
+            return
+        self._connecting = True
+        self._connected = False
+        _LOGGER.debug("[%s] Connecting to Mikrotik device", self._host)
+
+        kwargs = {
+            "encoding": self._encoding,
+            "login_methods": self._login_method,
+            "port": self._port,
+        }
+
+        if self._use_ssl:
+            ssl_context = ssl.create_default_context()
+            ssl_context.check_hostname = False
+            ssl_context.verify_mode = ssl.CERT_NONE
+            kwargs["ssl_wrapper"] = ssl_context.wrap_socket
+
+        try:
+            self._client = librouteros.connect(
+                self._host, self._user, self._password, **kwargs
+            )
+        except (
+            librouteros.exceptions.TrapError,
+            librouteros.exceptions.MultiTrapError,
+            librouteros.exceptions.ConnectionError,
+        ) as api_error:
+            _LOGGER.error("Mikrotik %s: %s", self._host, api_error)
+            self._connecting = False
+            self._client = None
+            return False
+
+        self.get_hostname()
+        if not self._host_name:
+            _LOGGER.error("Mikrotik failed to connect to %s", self._host)
+            return False
+        _LOGGER.info("Mikrotik Connected to %s (%s)", self._host_name, self._host)
+        self._connecting = False
+        self._connected = True
+        return True
+
+    def get_hostname(self):
+        """Return device host name."""
+        if not self._connected:
+            self.connect_to_device()
+        if not self._host_name:
+            cmd = MIKROTIK_SERVICES[IDENTITY]
+            self._host_name = (self._client(cmd=cmd))[0]["name"]
+        return self._host_name
+
+    def connected(self):
+        """Return connected boolean."""
+        return self._connected
+
+    def update_info(self):
+        """Update device info from Mikrotik API."""
+        data = self.command(MIKROTIK_SERVICES[INFO])
+        if data is None:
+            _LOGGER.error("Mikrotik device %s is not connected", self._host)
+            self.connect_to_device()
```

Comment:
We already try to connect in `self.command`.

Reply:
Removing connect_to_device()